torch_geometric/utils/augmentation.py

```diff
@@ -0,0 +1,204 @@
+from typing import Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from .num_nodes import maybe_num_nodes
+
+
+def shuffle_node(x: Tensor, training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly shuffle the feature matrix :obj:`x` along the
+    first dimmension.
+
+    The method returns (1) the shuffled :obj:`x`, (2) the permutation
+    indicating the orders of original nodes after shuffling.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    :rtype: (:class:`FloatTensor`, :class:`LongTensor`)
+
+    Example:
+
+        >>> x = torch.tensor([[0, 1, 2],
+        ...                   [3, 4, 5],
+        ...                   [6, 7, 8]], dtype=torch.float)
+        >>> x, node_perm = shuffle_node(x)
+        >>> x
+        tensor([[6., 7., 8.],
+                [3., 4., 5.],
+                [0., 1., 2.]])
+        >>> node_perm
+        tensor([2, 1, 0])
+    """
+    if not training:
+        perm = torch.arange(x.size(0), device=x.device)
+        return x, perm
+    perm = torch.randperm(x.size(0), device=x.device)
+    return x[perm], perm
+
+
+def mask_feature(x: Tensor, p: float = 0.5, mode: str = 'col',
+                 training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly masks feature from the feature matrix
+    :obj:`x` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`x`, (2) the feature
+    mask broadcastable with :obj:`x` (mode='row' and mode='col') or
+    with the same shape as :obj:`x` (mode='all'), indicating where features
+    are retained.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        p (float, optional): The masking ratio. (default: :obj:`0.5`)
+        mode (str, optional): The masked scheme to use for feature masking.
```

Comment:
Can we add more documentation to `mode`? For example:
If mode=col, will mask entire features of all nodes from the feature matrix. If mode=row, will mask entire nodes from the feature matrix. If mode=all, will mask individual features across all nodes.

Reply:
Got it.